parachain/primitives/core/src/outbound.rs

```diff
@@ -225,6 +235,14 @@ impl AgentExecuteCommand {
 					Token::Uint(U256::from(*amount)),
 				])),
 			]),
+			AgentExecuteCommand::Transact { target, payload, dynamic_gas } => ethabi::encode(&[
+				Token::Uint(self.index().into()), // TODO not sure what this does
```

Comment:
It's the abi encode of the Transact Enum matching what you defined above in solidity types.

Reply:
@yrong I'm specifically wondering what the `index` does?

Reply:
Just the enum index in solidity as you defined.
https://github.com/Snowfork/snowbridge/blob/5b87c0615c7b2e11cd77727b1ddde5b2b988c421/contracts/src/Types.sol#L76